src/neural/loader.cc

```diff
@@ -140,6 +209,37 @@ void FixOlderWeightsFile(WeightsFile* file) {
       net->set_smolgen_activation(pblczero::NetworkFormat::ACTIVATION_SWISH);
     }
   }
+
+  // Get updated network format.
+  network_format = file->format().network_format().network();
+  auto embedding_type = file->format().network_format().input_embedding();
+  bool multihead_format = (network_format & 128) == 128;
+  if (!multihead_format && network_format != nf::NETWORK_ONNX) {
+    auto weights = file->weights();
+    if (weights.has_policy_heads() && weights.has_value_heads()) {
+      CERR << "Weights file has multihead format, updating format flag";
+      net->set_network(static_cast<pblczero::NetworkFormat::NetworkStructure>(network_format | 128));
```

Comment:
Also it's better to have a helper function instead of `|128`

Reply:
Also dealt with in #1915